### ISSUE TITLE
help text doesn't work properly in parameters with underscore

### ISSUE THREAD

**Issue by iago1460** [pyinvoke/invoke#409]
The next code doesn't generate a help text:
```
@task(help={'include_author': 'Include author in the footer'})
def docs(ctx, include_author=False):
    """
    Docs generator
    """
    print("Documenting...")
```

```
$ inv --help docs
Usage: inv[oke] [--core-opts] docs [--options] [other tasks here ...]

Docstring:
  Docs generator

Options:
  -i, --include-author
```

But replacing the key "include_author" with "include-author" in the help dictionary works:
```
$ inv --help docs
Usage: inv[oke] [--core-opts] docs [--options] [other tasks here ...]

Docstring:
  Docs generator

Options:
  -i, --include-author   Include author in the footer
```

**Comment by bitprophet** [comment]
Nice catch, thanks. Sounds like a bug. Hopefully trivially fixed.

Slightly related to #398 as well, suspect both are fixable in the same spot or nearby spots.

**Comment by florisla** [comment]
#533 is a duplicate of this issue.  533 has a reference to here, but not the other way around (yet).

**Comment by kevin-dimichel-dr** [comment]
Help also fails to recognize functions if their name contains an underscore:

```python
@task
def docs(ctx, include_author=False):
    """
    Docs generator
    """
    print("Documenting...")

@task
def gen_docs(ctx, include_author=False):
    """
    Docs generator
    """
    print("Documenting...")
```

```shell
$ invoke --help docs
Usage: inv[oke] [--core-opts] docs [--options] [other tasks here ...]

Docstring:
  Docs generator

Options:
  -i, --include-author

$ invoke --help gen_docs
No idea what 'gen_docs' is!
```

**Comment by bitprophet** [comment]
Fixed in #611.
